### DISCUSSION TITLE
1 server entry, multiple access methods (ssh, sftp, ...)

### DISCUSSION THREAD

**Discussion by Thunderhawk2001** [1Remote/1Remote#525] [Ideas]
It would be pretty cool to have the option to access a server via ssh, sftp, but only have 1 entry for the server in the list.
e.g. I press "ALT + M", select a server, press "right arrow" and can select "Connect SSH" "Connect SFTP" "Connect....". Maybe extend this by having the option to press "ALT + M", select a server, press a user defined shortcut e.g. "F1" or "CTRL + 1" for SSH, "F2" or "CTRL + 2" for SFTP.... and so on.

Hope this makes any sense...  Coffee is still not working today and brain is slow.😪

**Comment by majkinetor** [comment]
You can currently have alternatives that can do that. Not all the options are supported, but since we have port there and credentials, it should work.

![image](https://github.com/1Remote/1Remote/assets/85767/4818287c-3def-46ed-aa93-6913722b0da0)

![image](https://github.com/1Remote/1Remote/assets/85767/b58102e7-d12e-4bc2-90b1-ef7b4563f6f8)

To have all of the connection options is IMO not something many people would need.

The alternatives show up in a menu, use right click in launcher.

Here is an example, that actually uses different servers with the same creds:

![image](https://github.com/1Remote/1Remote/assets/85767/c70ac571-42db-4cde-b77d-e4e709c36b39)

**Reply by Thunderhawk2001** [reply]
First of all: Thank you for your fast reply and creating this discussion.

Ah, so I'd have to "group" my SSH under one entry and my SFTP under another?
Wouldn't it be better, or maybe an option, to go like this:

I'd create an entry for "mylittleserver.com" with the IP "123.123.123.123". Then go to the settings of this entry and enter my SSH, SFTP, RDP, etc. account details, ports, certificates, etc.  Now I can press ALT + M, search for "mylittleserver", then right click, or press the right arrow key, and select the type of connection I want to establish: "Connect via SSH", "Connect via SFTP", "Connect via RDP".

So basically, group all different connection types under one server, not all servers under one connection type.

**Reply by majkinetor** [reply]
It indeed makes sense :) 

You proposition can be used as is, but perhaps something like this.

![image](https://github.com/1Remote/1Remote/assets/85767/2fc4e4fa-41db-4699-a3b4-b025cefd0a79)

where on + one would add any available protocol and when you change it, tab name takes its name. So no more all tabs visible. 

As I am not currently happy how runners are setup (#432) this could mean we could also add out of the box and custom protocols here.

One good example for me personally would be Windows would be RDP + PowerShell remote - I have that combination on all of my servers. 

/cc @VShawn

**Reply by majkinetor** [reply]
This would even allow adding more of the _same_ protocol. Several Apps or Telnets or Serials immediately come to mind.

**Reply by majkinetor** [reply]
I was thinking more about this, and while the idea IMO has merits, it certainly requires a lot of redesign

It would at least involve

1. Creating a generic tab page that holds host name, tags, icon, scripts.
3. Removing this generic stuff from all protocols
4. Creating interface for adding a connection, reconcile with methods of adding alternatives which are obsolete with these
5. It will certainly require providing some other UX elements
6. Change other interface elements that didn't take this into account, such as the launcher, which shows protocol next to connection name 
7. See what kind of limits we should have on protocols - what stops you from keeping ALL of your connections as a single entry for example ? Does it need to be prevented ? Who knows...

So this requires a lot of work.

In my mind, it looks like this:

![image](https://github.com/1Remote/1Remote/assets/85767/73844e5b-6caa-440e-93cc-7fb34dc534eb)

-  General tab (1) with common stuff. It probably needs some more things for better UX, such as providing default Hostname/Username/whatever for protocols in 3 and 5
- With + button (2), you add protocols. You can choose any available protocol and also the same one multiple times, which is why I removed _Alternatives_ (4), as adding another same protocol is more powerful than only 4 alternative fields we have now. If we add multiple same protocols we need name to distinguish them, just as we have now with alternatives. 
- (3) is a RDP session stripped from general stuff
- (5) is SSH session stripped from general stuff
- Some other things are needed here, like ordering of connections seems important (first one could be default for example), options to remove protocols, options to clone existing one as alternative etc.

It would be launched just as now with alternatives:

![image](https://github.com/1Remote/1Remote/assets/85767/18138479-9422-4d75-85df-590e77b03dfe)

I don't think we should do this at this phase, but it's a good idea anyway - I find that I would actually prefer this rather than have 2 or 3 entries for a single server - like I said, I always keep at least RDP and Powershell Remote App for my Windows servers, which makes some noise. Noise is easily solvable by tags though ( `#rdp -#powershell` for example) but this would provide reuse of common input elements, something you can't do now (common defaults).

/cc @VShawn

**Comment by VShawn** [comment]
So you mean to have one account password mapped to multiple protocols, instead of what @majkinetor mentioned about adding multiple account passwords (in fact, multiple account passwords are designed for the internal and external network accounts of the same server, I think @majkinetor would use it to store information for different servers :)). 

Currently, runner can meet your needs, which means adding an SFTP runner in SSH.

![image](https://github.com/1Remote/1Remote/assets/10143738/3e600511-9bdb-49b2-8137-4da5fd0bc242)


![image](https://github.com/1Remote/1Remote/assets/10143738/7f0488ad-0199-44f7-b6cf-a41d90060403)

![image](https://github.com/1Remote/1Remote/assets/10143738/35b9b29b-1af7-4424-8868-237a86b959b2)


unfortunate that I won't be able to further design for "one account password mapped to multiple protocols" because it completely contradicts the current data structure.

There might be other improvements in the future, but now I need stop thinking, still feeling sick and cough

**Reply by majkinetor** [reply]
No @VShawn, he wants one entry for multiple connections. 

Lets say you have RDP, SSH and VNC on the same server, he wants to have 1 entry and basically choose which protocol connection uses at invocation time, similar as with alternatives.

**Reply by VShawn** [reply]
`he wants one entry for multiple connections.`

YES thas's what I'm taling about: AS my screen shots show, I can connect my NAS via SSH or SFTP with 1 entry.

**Reply by majkinetor** [reply]
But this works only for one connection protocol, right? What if I want both RDP and SSH for example?
